### PR TITLE
Add custom dispatcher to builtin_string

### PR DESCRIPTION
**Pull request by orkvi**
JerryScript-DCO-1.0-Signed-off-by: Virag Orkenyi orkvi@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/ecma/builtin-objects/ecma-builtin-string.c [hunk 1]
```diff
@@ -387,6 +403,42 @@ ecma_builtin_string_dispatch_construct (const ecma_value_t *arguments_list_p, /*
...
+      return ecma_builtin_string_object_from_char_code (this_arg, arguments_list_p, arguments_number);
+    }
+#if JERRY_ESNEXT
+    case ECMA_BUILTIN_STRING_OBJECT_FROM_CODE_POINT:
+    {
+      return ecma_builtin_string_object_from_code_point (this_arg, arguments_list_p, arguments_number);
+    }
+    case ECMA_BUILTIN_STRING_OBJECT_RAW:
+    {
+      return ecma_builtin_string_object_raw (this_arg, arguments_list_p, arguments_number);
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtin-string.c:432]: argument this is all unused in all these three functions, and the behavior is same when no arguments is passed. This common part should be the part of the dispatcher

**Review by rerobika — CHANGES_REQUESTED** [review]

File: jerry-core/ecma/builtin-objects/ecma-builtin-string.c [hunk 2]
```diff
@@ -57,11 +73,9 @@
  *         Returned value must be freed with ecma_free_value.
  */
 static ecma_value_t
-ecma_builtin_string_object_from_char_code (ecma_value_t this_arg, /**< 'this' argument */
-                                           const ecma_value_t args[], /**< arguments list */
+ecma_builtin_string_object_from_char_code (const ecma_value_t args[], /**< arguments list */
                                            uint32_t args_number) /**< number of arguments */
 {
-  JERRY_UNUSED (this_arg);
```
Comment (zherczeg) [inline jerry-core/ecma/builtin-objects/ecma-builtin-string.c:64]: Remove newline below.

File: jerry-core/ecma/builtin-objects/ecma-builtin-string.c [hunk 3]
```diff
@@ -120,11 +134,9 @@ ecma_builtin_string_object_from_char_code (ecma_value_t this_arg, /**< 'this' ar
  *         Returned value must be freed with ecma_free_value.
  */
 static ecma_value_t
-ecma_builtin_string_object_raw (ecma_value_t this_arg, /**< 'this' argument */
-                                const ecma_value_t args[], /**< arguments list */
+ecma_builtin_string_object_raw (const ecma_value_t args[], /**< arguments list */
                                 uint32_t args_number) /**< number of arguments */
 {
-  JERRY_UNUSED (this_arg);
```
Comment (zherczeg) [inline jerry-core/ecma/builtin-objects/ecma-builtin-string.c:127]: Ditto.

File: jerry-core/ecma/builtin-objects/ecma-builtin-string.c [hunk 4]
```diff
@@ -387,6 +397,43 @@ ecma_builtin_string_dispatch_construct (const ecma_value_t *arguments_list_p, /*
...
+ *         Returned value must be freed with ecma_free_value.
+ */
+ecma_value_t
+ecma_builtin_string_dispatch_routine (uint8_t builtin_routine_id, /**< built-in wide routine identifier */
+                                      ecma_value_t this_arg, /**< 'this' argument value */
+                                      const ecma_value_t arguments_list_p[], /**< list of arguments
+                                                                              *   passed to routine */
+                                      uint32_t arguments_number) /**< length of arguments' list */
+{
+  JERRY_UNUSED (this_arg);
```
Comment (zherczeg) [inline jerry-core/ecma/builtin-objects/ecma-builtin-string.c:413]: Put a newline after this line.

**Review by zherczeg — CHANGES_REQUESTED** [review]
Only minor things.

File: jerry-core/ecma/builtin-objects/ecma-builtin-string.c [hunk 5]
```diff
@@ -274,11 +284,9 @@ ecma_builtin_string_object_raw (ecma_value_t this_arg, /**< 'this' argument */
  *         Returned value must be freed with ecma_free_value.
  */
 static ecma_value_t
-ecma_builtin_string_object_from_code_point (ecma_value_t this_arg, /**< 'this' argument */
-                                            const ecma_value_t args[], /**< arguments list */
+ecma_builtin_string_object_from_code_point (const ecma_value_t args[], /**< arguments list */
                                             uint32_t args_number) /**< number of arguments */
 {
-  JERRY_UNUSED (this_arg);
```
Comment (zherczeg) [inline jerry-core/ecma/builtin-objects/ecma-builtin-string.c:290]: Remove this newline.

**Review by zherczeg — CHANGES_REQUESTED** [review]

**Review by zherczeg — APPROVED** [review]
LGTM

**Review by galpeter — APPROVED** [review]
lgtm

**Review by rerobika — APPROVED** [review]

**Pull request merged**